Lib/unittest/mock.py

```diff
@@ -68,21 +69,22 @@ def _get_signature_object(func, as_instance, eat_self):
     signature object.
     Return a (reduced func, signature) tuple, or None.
     """
-    if isinstance(func, type) and not as_instance:
-        # If it's a type and should be modelled as a type, use __init__.
-        try:
-            func = func.__init__
-        except AttributeError:
-            return None
-        # Skip the `self` argument in __init__
-        eat_self = True
-    elif not isinstance(func, FunctionTypes):
-        # If we really want to model an instance of the passed type,
-        # __call__ should be looked up, not __init__.
-        try:
-            func = func.__call__
-        except AttributeError:
-            return None
+    if not isinstance(func, functools.partial):
```

Comment:
Just a suggestion, to follow the way the code is written, what do you think about:

If is partial:
  Pass
Elif is a class:
  ...

Reply:
I wrote it this way initially but I will be simply adding `pass` or `func = func` like a no op inside the clause and adding `if not` this way seemed reasonable but I too felt while writing since it kind of seems to make the diff large indicating larger change as in skipping a lot of code. 

```
if isinstance(func, functools.partial):
    pass
```

Reply:
As you wish, it was just a suggestion as that way it could read as each branch is a different type.

Reply:
I have changed to the suggested branching. Yours looks more inline with the flow 👍  I think adding a comment on no-op would help. How about "For partial objects use the underlying function signature instead of using constructor's signature as below"?